File: php/src/Behavioral/Visitor/TitleShortBlurbVisitor.php
<?

require_once 'BookInfo.php';
require_once 'GameInfo.php';
require_once 'DvdInfo.php';
require_once 'TitleBlurbVisitor.php';


class TitleShortBlurbVisitor extends TitleBlurbVisitor
{
	function visit( AbstractTitleInfo &$info )
	{
		$cname = get_class($info);
		if( $cname == 'DvdInfo' )
		{
			$this->setTitleBlurb( "SB-DVD: " .  $info->getTitleName() );
		}

		else if( $cname == 'BookInfo')
		{
			$this->setTitleBlurb( "SB-Book: " . $info->getTitleName() );
		}
		else if( $cname == 'GameInfo' )
		{
			$this->setTitleBlurb( "SB-Game: " . $info->getTitleName());
		}
	}
}

File: go/behavioral/visitor/TitleShortBlurbVisitor.go
package visitor

type TitleShortBlurbVisitor struct { // implements Visitor
	Title
}

func NewTitleShortBlurbVisitor () * TitleShortBlurbVisitor {
   return &TitleShortBlurbVisitor{&TitleInfo{}}
}

func ( this *TitleShortBlurbVisitor) Visit( info BlurberAcceptor ) {
	this.SetTitle( info.GetShortBlurb() )
}

